### PR TITLE
chore(components): Add sample for Sagemaker Model Monitor component

### PR DESCRIPTION
**Pull request by rd-pong**
**Description of your changes:**
- Add sample pipeline for Sagemaker Model Monitor component
- Add doc and data for sample pipeline
- Update sample pipeline main doc

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: samples/contrib/aws-samples/hosting_model_monitor_pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+### Create an IAM Role
+Follow [SageMaker execution role](../README.md#inputs-to-the-pipeline) and create an IAM role for SageMaker execution.
+
+### Create an S3 Bucket
+To setup an endpoint and create a monitoring schedule, we need an S3 bucket to store model and baseline data. Run the following commands to create an S3 bucket. Specify the value for `SAGEMAKER_REGION` as the region you want to create your SageMaker resources. For ease of use in the samples (using the default values of the pipeline), we suggest using `us-east-1` as the region.
+
+```
+export SAGEMAKER_REGION=us-east-1
+export S3_BUCKET_NAME="data-bucket-${SAGEMAKER_REGION}-$RANDOM"
```
Comment (ananth102) [inline samples/contrib/aws-samples/hosting_model_monitor_pipeline/README.md:18]: RANDOM's range is between 0-32767, can you change the prefix to "kfp-sm-data-bucket" to reduce the chance of collisions.

**Review by ananth102 — COMMENTED** [review]
looks good, small suggestions

File: samples/contrib/aws-samples/hosting_model_monitor_pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+else
+    aws s3api create-bucket --bucket ${S3_BUCKET_NAME} --region ${SAGEMAKER_REGION} \
+    --create-bucket-configuration LocationConstraint=${SAGEMAKER_REGION}
+fi
+
+echo ${S3_BUCKET_NAME}
+```
+
+### Copy data to bucket
+Fill the S3 bucket you just created with [sample data](./model-monitor/) which contains:
```
[the remark this answers is not in the document]
Comment (rd-pong) [reply samples/contrib/aws-samples/hosting_model_monitor_pipeline/README.md:31]: The model file is available in public S3 and we are getting it from the public bucket. But the baseline constraints and statistics are not available.

File: samples/contrib/aws-samples/hosting_model_monitor_pipeline/hosting_model_monitor_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+
+import kfp
+import sagemaker
```
[the remark this answers is not in the document]
Comment (rd-pong) [reply samples/contrib/aws-samples/hosting_model_monitor_pipeline/hosting_model_monitor_pipeline.py:4]: Added as requirements.txt

File: samples/contrib/aws-samples/hosting_model_monitor_pipeline/hosting_model_monitor_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+# Fetch environment variables
+SAGEMAKER_EXECUTION_ROLE_ARN = os.getenv("SAGEMAKER_EXECUTION_ROLE_ARN", "")
+S3_BUCKET_NAME = os.getenv("S3_BUCKET_NAME", "")
+SAGEMAKER_REGION = os.getenv("SAGEMAKER_REGION", "us-east-1")
+
+name_prefix = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+
+# Parameters for Model
+xgboost_image = sagemaker.image_uris.retrieve(
+    framework="xgboost", region=SAGEMAKER_REGION, version="0.90-1"
```
[the remark this answers is not in the document]
Comment (rd-pong) [reply samples/contrib/aws-samples/hosting_model_monitor_pipeline/hosting_model_monitor_pipeline.py:39]: We are keeping the same case with [sagemaker-example](https://sagemaker-examples.readthedocs.io/en/latest/sagemaker_model_monitor/introduction/SageMaker-ModelMonitoring.html)

**Pull request merged**